tests/purchase_test.ts

```diff
@@ -2,38 +2,52 @@ import LoginPage from "../pages/loginPage";
 import ProductPage from "../pages/productPage";
 import CartPage from "../pages/cartPage";
 import CheckoutPage from "../pages/checkoutPage";
-import User from "../helpers/interfaces";
+import User, { Product } from "../helpers/interfaces";
+import productPage from "../pages/productPage";
+import cartPage from "../pages/cartPage";
+var assert = require('assert');
 
-Feature('Purchase');
+const products = readAndParseData();
 
-let products = new DataTable(['url', 'color', 'size']);
-products.add(['/index.php?route=product/product&product_id=49', 'Green', 'Medium']);
-products.add(['/index.php?route=product/product&product_id=74', 'Black', null]); //checkout failure?
-products.add(['/index.php?route=product/product&product_id=73', null, null]);
-products.add(['/index.php?route=product/product&product_id=48', 'Green', 'Large']);
-products.add(['/index.php?route=product/product&product_id=43', 'Brown', null]); //options are absent
-products.add(['/index.php?route=product/product&product_id=47', 'White', null]); //checkout failure?
-products.add(['/index.php?route=product/product&product_id=60', null, null]); //checkout failure?
+Feature('Purchase');
 
 const USER: User = {
     email: `anastasiia0.2687341244603123@mailinator.com`,
     password: "50p8c0ze"
 };
 
-Before(async ({ I }) => { // or Background
-    LoginPage.login(USER);
+Before(async ({ I }) => {
+    await LoginPage.login(USER);
     I.see('My Affiliate Account');
     await CartPage.clearCart();
 });
 
 Data(products).Scenario('Buy 1 product',  async ({ I, current }) => {
     I.amOnPage(current.url);
     I.see('Product Code');
-    await ProductPage.selectOption(current.color, current.size);
+    let price: number = parseFloat((await I.grabTextFrom(productPage.priceLabel)).split('$')[1]);
+    price = price + await ProductPage.selectOption(current.color, current.size);
     ProductPage.addToCart();
     ProductPage.openCart();
-    I.see("My Cart")
-    await CartPage.proceedToCheckout();
-    await CheckoutPage.placeOrder();
-    I.see("Your order has been placed!");
-});
\ No newline at end of file
+    await I.see("My Cart");
+    if (await I.grabNumberOfVisibleElements(cartPage.productUnavailableLabel)){
+        console.log("This product is unavailable for purchase.");
+    } else {
+        await CartPage.proceedToCheckout();
+        let checkoutPrice: number = await CheckoutPage.placeOrder();
+        assert.equal(price, checkoutPrice);
+        I.see("Your order has been placed!");
+    }
+});
+
+function readAndParseData(): CodeceptJS.DataTable {
```

Comment:
Import array from json(use new schema) and use it in `Data(products)`.
`import products from './json-file.json'`
Then this function is not required and makes your code more simple. But for practice it is OK.